webapp/src/Controller/API/GeneralInfoController.php

```diff
@@ -286,6 +288,48 @@ public function getConfigCheckAction(): JsonResponse
         return $this->json($result, $aggregate);
     }
 
+    /**
+     * Get the flag for the given country
+     * @Rest\Get("/country-flags/{countryCode}/{size}.svg")
+     * @OA\Response(
+     *     response="200",
+     *     description="Returns the given country flag in SVG format",
+     *     @OA\MediaType(mediaType="image/svg+xml")
+     * )
+     * @OA\Parameter(
+     *     name="countryCode",
+     *     in="path",
+     *     description="The ISO 3166-1 alpha-3 code for the country to get the flag for",
+     *     @OA\Schema(type="string")
+     * )
+     */
+    public function countryFlagAction(string $countryCode, string $size): Response
+    {
+        // This API action exists for two reasons
+        // - Relative URL's are relative to the API root according to the CCS spec. This
+        //   means that we need to have an API endpoint for files.
+        // - This makes it that we can not return a flag if flags are disabled.
+
+        if (!$this->config->get('show_flags')) {
+            throw new NotFoundHttpException('country flag not found');
+        }
+
+        $alpha3code = strtoupper($countryCode);
+        if (!Countries::alpha3CodeExists($alpha3code)) {
+            throw new NotFoundHttpException('country flag not found');
+        }
+        $alpha2code = strtolower(Countries::getAlpha2Code($alpha3code));
+        $flagFile = sprintf('%s/public/flags/%s/%s.svg', $this->dj->getDomjudgeWebappDir(), $size, $alpha2code);
+
+        if (!file_exists($flagFile)) {
+            throw new NotFoundHttpException('country flag not found');
```

Comment:
How is this different from the one above?

Reply:
If we somehow not have an SVG for a valid alpha3/2 code